benchmarks/src/lib.rs

```diff
@@ -15,4 +15,116 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use datafusion::DATAFUSION_VERSION;
+use serde::Serialize;
+use serde_json::Value;
+use std::{collections::HashMap, time::SystemTime};
+
 pub mod tpch;
+
+#[derive(Debug, Serialize)]
+pub struct RunContext {
+    /// Benchmark crate version
+    pub benchmark_version: String,
+    /// DataFusion crate version
+    pub datafusion_version: String,
+    /// Number of CPU cores
+    pub num_cpus: usize,
+    /// Start time
+    pub start_time: u64,
+    /// CLI arguments
+    pub arguments: Vec<String>,
+}
+
+impl Default for RunContext {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl RunContext {
+    pub fn new() -> Self {
+        Self {
+            benchmark_version: env!("CARGO_PKG_VERSION").to_owned(),
+            datafusion_version: DATAFUSION_VERSION.to_owned(),
+            num_cpus: num_cpus::get(),
+            start_time: SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .expect("current time is later than UNIX_EPOCH")
+                .as_secs(),
+            arguments: std::env::args().skip(1).collect::<Vec<String>>(),
+        }
+    }
+}
+
+/// A single iteration of a benchmark query
+#[derive(Debug, Serialize)]
+struct QueryIter {
+    elapsed: f64,
```

Comment:
Can you please add some documentation about what unit this is in (I think it is milliseconds?)

Relatedly I wonder if we could make this API easier to use by storing a `Duration` https://doc.rust-lang.org/std/time/struct.Duration.html, calculated with `SystemTime::now() - start`

Reply:
I've changed `elapsed` to be a `Duration` object